sui-execution/latest/sui-move-natives/src/dynamic_field.rs

```diff
@@ -231,6 +232,13 @@ pub fn add_child_object(
             * struct_tag_size.into()
     );
 
+    if object_is_shared(context, &child)? {
```

Comment:
Why is this check necessary and why wasn't it needed prior to this PR?

Reply:
`dynamic_field::add` accepts the field by value, so it's the fact that this PR allows taking shared objects by value that introduces the requirement for this check.